### ISSUE TITLE
Implement array-selector

### ISSUE THREAD

**Issue by blasten** [Polymer/polymer#4154]
`<array-selector>` is a dependency of iron-list.

**Comment by Alos** [comment]
Is this related to my issue in [iron-list](https://github.com/PolymerElements/iron-list/issues/353)?

**Comment by blasten** [comment]
@Alos That's right

**Comment by Alos** [comment]
Interesting! What are you guys thinking? Should `<array-selector>` be a behavior or maybe temporarily spawn `<array-selector>` as a util element and make it available for other elements that use it \*cough\* _iron-list_ \*cough\*? It might be a little over my head, but I could try to help?

**Comment by kevinpschaaf** [comment]
@Alos Thanks for the offer Alos... `array-selector` in particular is affected by our proposed changes to how [object dirty checking works](https://github.com/Polymer/polymer/blob/2.0-preview/README.md#breaking-data-dirty-checking). I went ahead and did a first-pass implementation to work out the kinks that we'll be reviewing shortly.  I did go ahead and make it a mixin for more lightweight integration.
